index.js

```
@@ -85,88 +85,120 @@ function getWeights() {
 			100;
 		weights[group] = weight;
 	}
-
+	console.log(weights);
 	return weights;
 }
 
 function getAssignments() {
 	const assignments = [];
+	// Use a broader selector to include all assignment rows,
+	// not just those already marked as graded.
+	const assignmentRows = document.querySelectorAll(
+		'#grades_summary tr.student_assignment'
+	);
 
-	// Scrape assignments table for graded assignments.
-	for (const assignment of document.querySelectorAll(
-		'#grades_summary tr.assignment_graded.student_assignment'
-	)) {
-		let earned, available, title, group;
-
-		const a = assignment.querySelector('th.title');
-		title = a.querySelector('a').textContent;
-		group = a.querySelector('div.context').textContent;
-
-		const due = parseCanvasDate(
-			assignment.querySelector('.due').textContent.trim()
-		);
-		const submitted = parseCanvasDate(
-			assignment.querySelector('.submitted').textContent.trim()
-		);
-
-		const grades = assignment.querySelector('td.assignment_score span.grade');
-
-		// Submitted but not yet graded
-		if (grades.querySelector('.submission_icon')) continue;
-
-		// Complete or incomplete
-		if (grades.querySelector('.graded_icon')) {
-			earned = Number.parseFloat(
-				assignment.querySelector('.original_points').textContent.trim()
-			);
+	for (const row of assignmentRows) {
+		let earned = 0,
+			available = 0,
+			title = '',
+			group = '';
+
+		// Get assignment title and group from the header cell.
+		const titleEl = row.querySelector('th.title');
+		if (!titleEl) continue;
```

```suggestion
```
I'd rather throw so we know something happened rather than silently ignoring it.